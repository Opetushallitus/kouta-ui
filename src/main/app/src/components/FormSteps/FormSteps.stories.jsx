import React from 'react';

import { storiesOf } from '@storybook/react';

import FormSteps from './index';

storiesOf('FormSteps', module).add('Basic', () => (
  <FormSteps activeStep="hakukohde" />
));
