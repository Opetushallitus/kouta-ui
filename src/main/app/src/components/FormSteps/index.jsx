import React from 'react';
import styled from 'styled-components';

import FormStepsIcon from './FormStepsIcon';

const GroupContainer = styled.div`
  display: flex;
`;

const StepsIcon = styled(FormStepsIcon)`
  flex: 0;
  margin-right: ${({ theme }) => theme.spacing.unit * 3}px;

  &:last-child {
    margin-right: 0px;
  }
`;

const Wrapper = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-between;
`;

const stepOrder = [
  'koulutus',
  'toteutus',
  'haku',
  'hakukohde',
  'valintaperusteet',
];

const stepIsActive = (step, activeStep) => {
  return step === activeStep;
};

const stepIsDone = (step, activeStep) => {
  const activeStepIndex = stepOrder.indexOf(activeStep);
  const stepIndex = stepOrder.indexOf(step);

  return stepIndex < activeStepIndex;
};

const FormSteps = ({ activeStep = 'koulutus', ...props }) => (
  <Wrapper {...props}>
    <GroupContainer>
      <StepsIcon
        icon="school"
        active={stepIsActive('koulutus', activeStep)}
        done={stepIsDone('koulutus', activeStep)}
      >
        Koulutus
      </StepsIcon>
      <StepsIcon
        icon="settings"
        active={stepIsActive('toteutus', activeStep)}
        done={stepIsDone('toteutus', activeStep)}
      >
        Toteutus
      </StepsIcon>
    </GroupContainer>
    <GroupContainer>
      <StepsIcon
        icon="access_time"
        active={stepIsActive('haku', activeStep)}
        done={stepIsDone('haku', activeStep)}
      >
        Haku
      </StepsIcon>
      <StepsIcon
        icon="grain"
        active={stepIsActive('hakukohde', activeStep)}
        done={stepIsDone('hakukohde', activeStep)}
      >
        Hakukohde
      </StepsIcon>
    </GroupContainer>
    <GroupContainer>
      <StepsIcon
        icon="select_all"
        active={stepIsActive('valintaperusteet', activeStep)}
        done={stepIsDone('valintaperusteet', activeStep)}
      >
        Valintaperusteet
      </StepsIcon>
    </GroupContainer>
  </Wrapper>
);

export default FormSteps;
