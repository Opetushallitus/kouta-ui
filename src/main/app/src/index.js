import 'core-js/features/object/keys';
import 'core-js/features/object/values';
import 'core-js/features/promise';
import 'core-js/features/array/includes';
import 'core-js/features/array/every';
import 'core-js/features/array/flat-map';

import React from 'react';
import ReactDOM from 'react-dom';
import createBrowserHistory from 'history/createBrowserHistory';
import { urls as ophUrls } from 'oph-urls-js';
import get from 'lodash/get';
import merge from 'lodash/merge';

import * as serviceWorker from './serviceWorker';
import App from './components/App';
import createStore from './state';
import defaultTheme from './theme';
import configureUrls from './apiUrls';
import createLocalisation from './localisation';
import { getLocalisation } from './apiUtils';
import getTranslations from './translations';
import createHttpClient from './httpClient';

const history = createBrowserHistory({ basename: 'kouta' });

serviceWorker.unregister();

const isDev = process.env.NODE_ENV === 'development';

const loadLocalisation = async ({
  namespace,
  language,
  httpClient,
  apiUrls,
}) => {
  const localisation = await getLocalisation({
    category: namespace,
    locale: language,
    httpClient,
    apiUrls,
  });

  const translations = getTranslations();

  return get(translations, [language, namespace])
    ? merge({}, translations[language][namespace], localisation || {})
    : localisation;
};

(async () => {
  const apiUrls = await configureUrls(ophUrls);

  const httpClient = createHttpClient({
    apiUrls,
    callerId: !isDev ? process.env.REACT_APP_CALLER_ID : null,
  });

  const localisationInstance = createLocalisation({
    debug: isDev,
    loadLocalisation: ({ namespace, language }) =>
      loadLocalisation({ namespace, language, httpClient, apiUrls }),
  });

  window.__i18n__ = localisationInstance;

  const { store, persistor } = createStore({
    apiUrls,
    httpClient,
    history,
    localisation: localisationInstance,
  });

  window.__store__ = store;

  ReactDOM.render(
    <App
      store={store}
      theme={defaultTheme}
      urls={apiUrls}
      httpClient={httpClient}
      history={history}
      localisation={localisationInstance}
      persistor={persistor}
    />,
    document.getElementById('root'),
  );
})();
