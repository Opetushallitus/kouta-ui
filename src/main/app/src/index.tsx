import React from 'react';

import { Globals } from '@react-spring/web';
import { inspect } from '@xstate/inspect';
import { createBrowserHistory } from 'history';
import { urls as ophUrls } from 'oph-urls-js';
import ReactDOM from 'react-dom';

import createHttpClient from './httpClient';
import { createDefaultLocalization } from './localization';
import App from './pages/App';
import { createStore } from './state';
import defaultTheme from './theme';
import { configure as configureUrls } from './urls';
import { isPlaywright } from './utils';

Globals.assign({
  skipAnimation: isPlaywright,
});

if (import.meta.env.VITE_XSTATE_INSPECTOR) {
  inspect({
    iframe: false, // open in new window
  });
}

const history = createBrowserHistory({ basename: 'kouta' });

(async () => {
  let apiUrls = ophUrls;

  const httpClient = createHttpClient({
    apiUrls,
    callerId: import.meta.env.VITE_CALLER_ID,
  });

  apiUrls = await configureUrls(apiUrls, httpClient);

  const localizationInstance = await createDefaultLocalization({
    httpClient,
    apiUrls,
  });

  const { store, persistor } = createStore();

  ReactDOM.render(
    <App
      store={store}
      theme={defaultTheme}
      urls={apiUrls}
      httpClient={httpClient}
      history={history}
      localization={localizationInstance}
      persistor={persistor}
    />,
    document.getElementById('root')
  );
})();
