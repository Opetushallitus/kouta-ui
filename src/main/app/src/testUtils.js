import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import defaultTheme from './theme';

const makeFnWithTheme = fn => tree => {
  return fn(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>);
};

export const shallowWithTheme = makeFnWithTheme(shallow);

export const mountWithTheme = makeFnWithTheme(mount);

export const setMockDate = date => {
  const _Date = Date;

  global.Date.now = () => +date;

  return () => {
    global.Date = _Date;
  };
};
