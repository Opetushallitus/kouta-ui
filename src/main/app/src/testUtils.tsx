import React, { PropsWithChildren } from 'react';

import { configureStore } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

import { createRootReducer } from './state/rootReducer';
import defaultTheme from './theme';

const makeFnWithTheme = fn => tree => {
  return fn(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>);
};

export const renderWithTheme = makeFnWithTheme(render);

export const mountWithTheme = tree => {
  const { container } = makeFnWithTheme(render)(tree);
  return container.firstChild;
};

export const setMockDate = date => {
  const _Date = Date;

  global.Date.now = () => Number(date);

  return () => {
    global.Date = _Date;
  };
};

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer: createRootReducer({}), preloadedState }),
    ...renderOptions
  }
) {
  // eslint-disable-next-line @typescript-eslint/ban-types
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return (
      <Provider store={store}>
        <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
      </Provider>
    );
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';
