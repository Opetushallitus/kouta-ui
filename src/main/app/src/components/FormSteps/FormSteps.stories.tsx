import React from 'react';

import FormSteps from './index';

export default {
  title: 'FormSteps',
};

export const Basic = () => <FormSteps activeStep="hakukohde" />;
