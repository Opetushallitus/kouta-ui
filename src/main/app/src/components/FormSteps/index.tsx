import React from 'react';
import styled from 'styled-components';
import { useTranslation } from 'react-i18next';

import { ENTITY, ICONS } from '#/src/constants';
import FormStepIcon from '#/src/components/FormStepIcon';

const {
  KOULUTUS,
  TOTEUTUS,
  HAKU,
  HAKUKOHDE,
  VALINTAPERUSTE,
  SORA_KUVAUS,
} = ENTITY;

const GroupContainer = styled.div`
  display: flex;
`;

const StepsIcon = styled(FormStepIcon)`
  flex: 0;
  margin-right: ${({ theme }) => theme.spacing.unit * 3}px;

  &:last-child {
    margin-right: 0px;
  }
`;

const Wrapper = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-between;
`;

const stepOrder = [
  KOULUTUS,
  TOTEUTUS,
  HAKU,
  HAKUKOHDE,
  VALINTAPERUSTE,
  SORA_KUVAUS,
];

const stepIsActive = (step, activeStep) => {
  return step === activeStep;
};

const stepIsDone = (step, activeStep) => {
  const activeStepIndex = stepOrder.indexOf(activeStep);
  const stepIndex = stepOrder.indexOf(step);

  return stepIndex < activeStepIndex;
};

const FormSteps = ({ activeStep = KOULUTUS }) => {
  const { t } = useTranslation();

  return (
    <Wrapper>
      <GroupContainer>
        <StepsIcon
          icon={ICONS[KOULUTUS]}
          active={stepIsActive(KOULUTUS, activeStep)}
          done={stepIsDone(KOULUTUS, activeStep)}
        >
          {t('yleiset.koulutus')}
        </StepsIcon>
        <StepsIcon
          icon={ICONS[TOTEUTUS]}
          active={stepIsActive(TOTEUTUS, activeStep)}
          done={stepIsDone(TOTEUTUS, activeStep)}
        >
          {t('yleiset.toteutus')}
        </StepsIcon>
      </GroupContainer>
      <GroupContainer>
        <StepsIcon
          icon={ICONS[HAKU]}
          active={stepIsActive(HAKU, activeStep)}
          done={stepIsDone(HAKU, activeStep)}
        >
          {t('yleiset.haku')}
        </StepsIcon>
        <StepsIcon
          icon={ICONS[HAKUKOHDE]}
          active={stepIsActive(HAKUKOHDE, activeStep)}
          done={stepIsDone(HAKUKOHDE, activeStep)}
        >
          {t('yleiset.hakukohde')}
        </StepsIcon>
      </GroupContainer>
      <GroupContainer>
        <StepsIcon
          icon={ICONS[VALINTAPERUSTE]}
          active={stepIsActive(VALINTAPERUSTE, activeStep)}
          done={stepIsDone(VALINTAPERUSTE, activeStep)}
        >
          {t('yleiset.valintaperusteet')}
        </StepsIcon>
        <StepsIcon
          icon={ICONS[SORA_KUVAUS]}
          active={stepIsActive(SORA_KUVAUS, activeStep)}
          done={stepIsDone(SORA_KUVAUS, activeStep)}
        >
          {t('yleiset.soraKuvaus')}
        </StepsIcon>
      </GroupContainer>
    </Wrapper>
  );
};

export default FormSteps;
