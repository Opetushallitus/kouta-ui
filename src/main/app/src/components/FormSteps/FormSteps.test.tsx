import { mountWithTheme } from '#/src/testUtils';

import FormSteps from './index';

test('renders correctly', () => {
  expect(
    mountWithTheme(<FormSteps activeStep="hakukohde" />)
  ).toMatchSnapshot();
});
