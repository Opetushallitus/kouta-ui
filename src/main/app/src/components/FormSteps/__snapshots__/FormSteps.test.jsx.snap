// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c6 {
  font-size: 1.5rem;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  border-radius: 50%;
  width: 3.5rem;
  height: 3.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 3px solid;
  background-color: white;
  border-color: #cecece;
  border-color: #2da0c7;
}

.c8 {
  border-radius: 50%;
  width: 3.5rem;
  height: 3.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 3px solid;
  background-color: white;
  border-color: #cecece;
  background-color: #2da0c7;
  border-color: white;
}

.c11 {
  border-radius: 50%;
  width: 3.5rem;
  height: 3.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 3px solid;
  background-color: white;
  border-color: #cecece;
}

.c7 {
  margin-top: 8px;
  text-align: center;
  font-family: 'Roboto',sans-serif;
  color: #666666;
  font-size: 0.8rem;
  white-space: nowrap;
}

.c10 {
  margin-top: 8px;
  text-align: center;
  font-family: 'Roboto',sans-serif;
  color: #666666;
  font-size: 0.8rem;
  white-space: nowrap;
  color: #2da0c7;
}

.c5 {
  font-size: 2rem;
  color: #666666;
}

.c9 {
  font-size: 2rem;
  color: #666666;
  color: white;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  margin-right: 24px;
}

.c2:last-child {
  margin-right: 0px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<styled.div>
  <div
    className="c0"
  >
    <styled.div>
      <div
        className="c1"
      >
        <Styled(FormStepIcon)
          active={false}
          done={true}
          icon="school"
        >
          <styled.div
            className="c2"
          >
            <div
              className="c2 c3"
            >
              <styled.div
                active={false}
                done={true}
              >
                <div
                  className="c4"
                >
                  <Styled(Component)
                    active={false}
                    type="school"
                  >
                    <ForwardRef
                      active={false}
                      className="c5"
                      type="school"
                    >
                      <styled.i
                        active={false}
                        className="material-icons c5"
                      >
                        <i
                          className="material-icons c5 c6"
                        >
                          school
                        </i>
                      </styled.i>
                    </ForwardRef>
                  </Styled(Component)>
                </div>
              </styled.div>
              <styled.div
                active={false}
              >
                <div
                  className="c7"
                >
                  yleiset.koulutus
                </div>
              </styled.div>
            </div>
          </styled.div>
        </Styled(FormStepIcon)>
        <Styled(FormStepIcon)
          active={false}
          done={true}
          icon="settings"
        >
          <styled.div
            className="c2"
          >
            <div
              className="c2 c3"
            >
              <styled.div
                active={false}
                done={true}
              >
                <div
                  className="c4"
                >
                  <Styled(Component)
                    active={false}
                    type="settings"
                  >
                    <ForwardRef
                      active={false}
                      className="c5"
                      type="settings"
                    >
                      <styled.i
                        active={false}
                        className="material-icons c5"
                      >
                        <i
                          className="material-icons c5 c6"
                        >
                          settings
                        </i>
                      </styled.i>
                    </ForwardRef>
                  </Styled(Component)>
                </div>
              </styled.div>
              <styled.div
                active={false}
              >
                <div
                  className="c7"
                >
                  yleiset.toteutus
                </div>
              </styled.div>
            </div>
          </styled.div>
        </Styled(FormStepIcon)>
      </div>
    </styled.div>
    <styled.div>
      <div
        className="c1"
      >
        <Styled(FormStepIcon)
          active={false}
          done={true}
          icon="access_time"
        >
          <styled.div
            className="c2"
          >
            <div
              className="c2 c3"
            >
              <styled.div
                active={false}
                done={true}
              >
                <div
                  className="c4"
                >
                  <Styled(Component)
                    active={false}
                    type="access_time"
                  >
                    <ForwardRef
                      active={false}
                      className="c5"
                      type="access_time"
                    >
                      <styled.i
                        active={false}
                        className="material-icons c5"
                      >
                        <i
                          className="material-icons c5 c6"
                        >
                          access_time
                        </i>
                      </styled.i>
                    </ForwardRef>
                  </Styled(Component)>
                </div>
              </styled.div>
              <styled.div
                active={false}
              >
                <div
                  className="c7"
                >
                  yleiset.haku
                </div>
              </styled.div>
            </div>
          </styled.div>
        </Styled(FormStepIcon)>
        <Styled(FormStepIcon)
          active={true}
          done={false}
          icon="grain"
        >
          <styled.div
            className="c2"
          >
            <div
              className="c2 c3"
            >
              <styled.div
                active={true}
                done={false}
              >
                <div
                  className="c8"
                >
                  <Styled(Component)
                    active={true}
                    type="grain"
                  >
                    <ForwardRef
                      active={true}
                      className="c9"
                      type="grain"
                    >
                      <styled.i
                        active={true}
                        className="material-icons c9"
                      >
                        <i
                          className="material-icons c9 c6"
                        >
                          grain
                        </i>
                      </styled.i>
                    </ForwardRef>
                  </Styled(Component)>
                </div>
              </styled.div>
              <styled.div
                active={true}
              >
                <div
                  className="c10"
                >
                  yleiset.hakukohde
                </div>
              </styled.div>
            </div>
          </styled.div>
        </Styled(FormStepIcon)>
      </div>
    </styled.div>
    <styled.div>
      <div
        className="c1"
      >
        <Styled(FormStepIcon)
          active={false}
          done={false}
          icon="select_all"
        >
          <styled.div
            className="c2"
          >
            <div
              className="c2 c3"
            >
              <styled.div
                active={false}
                done={false}
              >
                <div
                  className="c11"
                >
                  <Styled(Component)
                    active={false}
                    type="select_all"
                  >
                    <ForwardRef
                      active={false}
                      className="c5"
                      type="select_all"
                    >
                      <styled.i
                        active={false}
                        className="material-icons c5"
                      >
                        <i
                          className="material-icons c5 c6"
                        >
                          select_all
                        </i>
                      </styled.i>
                    </ForwardRef>
                  </Styled(Component)>
                </div>
              </styled.div>
              <styled.div
                active={false}
              >
                <div
                  className="c7"
                >
                  yleiset.valintaperusteet
                </div>
              </styled.div>
            </div>
          </styled.div>
        </Styled(FormStepIcon)>
        <Styled(FormStepIcon)
          active={false}
          done={false}
          icon="subject"
        >
          <styled.div
            className="c2"
          >
            <div
              className="c2 c3"
            >
              <styled.div
                active={false}
                done={false}
              >
                <div
                  className="c11"
                >
                  <Styled(Component)
                    active={false}
                    type="subject"
                  >
                    <ForwardRef
                      active={false}
                      className="c5"
                      type="subject"
                    >
                      <styled.i
                        active={false}
                        className="material-icons c5"
                      >
                        <i
                          className="material-icons c5 c6"
                        >
                          subject
                        </i>
                      </styled.i>
                    </ForwardRef>
                  </Styled(Component)>
                </div>
              </styled.div>
              <styled.div
                active={false}
              >
                <div
                  className="c7"
                >
                  yleiset.soraKuvaus
                </div>
              </styled.div>
            </div>
          </styled.div>
        </Styled(FormStepIcon)>
      </div>
    </styled.div>
  </div>
</styled.div>
`;
